app/src/main/java/org/mozilla/fenix/compose/button/Button.kt

```diff
@@ -31,55 +29,213 @@ import org.mozilla.fenix.theme.FirefoxTheme
 import org.mozilla.fenix.theme.Theme
 
 /**
- * Primary button.
+ * Base component for buttons.
  *
  * @param text The button text to be displayed.
- * @param icon Optional button icon to be displayed.
+ * @param contentColor [Color] to apply to the button text.
+ * @param backgroundColor The background [Color] of the button.
+ * @param icon Optional [Painter] used to display a [Icon] before the button text.
+ * @param tint Tint [Color] to be applied to the icon.
  * @param onClick Invoked when the user clicks on the button.
  */
 @Composable
-fun PrimaryButton(
+fun Button(
```

Comment:
should the base component be private?

Reply:
This is currently being used in `RecentSyncedTab` for the Button skeleton, but otherwise TBD.